x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java

```diff
@@ -302,9 +303,19 @@ private void fetchFollowerShardInfo(
             if (filteredShardStats.isPresent()) {
                 final ShardStats shardStats = filteredShardStats.get();
                 final CommitStats commitStats = shardStats.getCommitStats();
-                final String historyUUID = commitStats.getUserData().get(Engine.HISTORY_UUID_KEY);
-
+                if (commitStats == null) {
+                    // UnavailableShardsException will be retried byShardFollowNodeTask.shouldRetry(...)
+                    errorHandler.accept(new UnavailableShardsException(shardId, "commit_stats are missing"));
```

Comment:
Maybe throw "AlreadyClosedException" instead for commitStats or seqNoStats is null when we hit AlreadyClosedException. https://github.com/elastic/elasticsearch/blob/6500b0cbd78a0e3968ecb54a33b59a2325059ff9/server/src/main/java/org/elasticsearch/action/admin/cluster/stats/TransportClusterStatsAction.java#L110-L115